middleware/gzip/response_filter.go

```diff
@@ -62,11 +63,15 @@ func (r *ResponseFilterWriter) WriteHeader(code int) {
 	} else {
 		r.ResponseWriter.WriteHeader(code)
 	}
+	r.statusCodeWritten = true
 }
 
 // Write wraps underlying Write method and compresses if filters
 // are satisfied
 func (r *ResponseFilterWriter) Write(b []byte) (int, error) {
+	if !r.statusCodeWritten {
```

Comment:
This field isn't racy, right? (Just double-checking my logic - even though I think think this variable is only used in the same goroutine?)